indigo_app/tests/test_places.py

```diff
@@ -57,7 +57,8 @@ def test_place_settings(self):
         form['spreadsheet_url'].value = 'https://docs.google.com/spreadsheets/u/1/d/1a2o-842lGliSwlLo3gSbYSRbaOYu-2PZhC1rOf8MgA4/edit'
         form['as_at_date'].value = '2019-01-01'
         form['styleguide_url'].value = 'https://docs.laws.africa/editing-a-document/importing-a-document'
-        form = form.submit().follow().forms[0]
+        form = form.submit().forms[0]
```

Comment:
I'm not sure this is correct. If the form validates and saves successfully, the view will redirect back to the page, so the redirect must be followed to verify that the page updated correctly. If the form doesn't validate, then there will be no redirect.

So it seems to me like this test is actually failing because it's not redirecting when it should.

Reply:
To clarify:

Validation succeeds: POST -> redirect -> GET -> html page with updated details

Validation fails: POST -> response with html with failure messages

Reply:
Got it, thanks!